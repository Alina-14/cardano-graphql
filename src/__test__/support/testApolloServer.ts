import { ApolloServerBase } from 'apollo-server-core'
import * as fs from 'fs'
import * as path from 'path'
import * as depthLimit from 'graphql-depth-limit'
import { getConfig } from '../../config'

export async function testApolloServer (): Promise<ApolloServerBase> {
  process.env.HASURA_URI = 'http://localhost:8090/v1/graphql'
  const { context, resolvers } = await getConfig()
  return new ApolloServerBase({
    context,
    introspection: true,
    resolvers,
    typeDefs: fs.readFileSync(path.join(__dirname, '../../schema.graphql'), 'UTF8'),
    validationRules: [depthLimit(20)]
  })
}