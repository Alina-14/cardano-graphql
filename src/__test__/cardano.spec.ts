import gql from 'graphql-tag'
import { createTestClient, ApolloServerTestClient } from 'apollo-server-testing'
import { testApolloServer } from './support/testApolloServer'

describe('cardano', () => {
  let client: ApolloServerTestClient

  beforeEach(async () => {
    client = createTestClient(await testApolloServer())
  }, 60000)

  it('Returns key information about the network', async () => {
    const result = await client.query({
      query: gql`query {
          cardano {
              blockHeight
              currentEpoch {
                  number
              }
              protocolConst
              slotDuration
              startTime
          }
      }`
    })
    expect(result).toMatchSnapshot()
  })

})
