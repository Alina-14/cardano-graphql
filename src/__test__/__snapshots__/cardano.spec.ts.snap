// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cardano Returns key information about the network 1`] = `
Object {
  "data": Object {
    "cardano": Object {
      "blockHeight": 31070,
      "currentEpoch": Object {
        "number": 1,
      },
      "protocolConst": 2160,
      "slotDuration": 20000,
      "startTime": "2017-09-23T21:44:51",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
